ShareTheTable/View Controllers/Profile Making/Question/QuestionViewController.m

```diff
@@ -0,0 +1,182 @@
+//
+//  QuestionViewController.m
+//  ShareTheTable
+//
+//  Created by Andrea Gonzalez on 7/7/22.
+//
+
+#import "QuestionViewController.h"
+#import "AddPhotosViewController.h"
+#import "SceneDelegate.h"
+#import "FeedViewController.h"
+#import "Parse/Parse.h"
+
+@interface QuestionViewController ()
+
+@end
+
+@implementation QuestionViewController
+
+- (void)refreshButtons:(UIButton*)button {
+    // Button was already selected
+    if(button.selected == YES) {
+        button.selected = NO;
+        [button setBackgroundColor:UIColor.clearColor];
+    } else {
+        // Button was not already selected
+        button.selected = YES;
+        [button setBackgroundColor:UIColor.greenColor];
+    }
+}
+
+- (IBAction)didTapChineseFood:(id)sender {
+    [self refreshButtons:self.chineseFoodButton];
+}
+
+- (IBAction)didTapMexicanFood:(id)sender {
+    [self refreshButtons:self.mexicanFoodButton];
+}
+
+- (IBAction)didTapAmericanFood:(id)sender {
+    [self refreshButtons:self.americanFoodButton];
+}
+
+- (IBAction)didTapItalianFood:(id)sender {
+    [self refreshButtons:self.italianFoodButton];
+}
+
+- (IBAction)didTapGreekFood:(id)sender {
+    [self refreshButtons:self.greekFoodButton];
+}
+
+- (IBAction)didTapOtherFood:(id)sender {
+    [self refreshButtons:self.otherFoodButton];
+}
+
+- (IBAction)didTapSeaFoodAllergies:(id)sender {
+    [self refreshButtons:self.seaFoodAllergiesButton];
+}
+
+- (IBAction)didTapLactoseIntolerant:(id)sender {
+    [self refreshButtons:self.lactoseIntolerantButton];
+}
+
+- (IBAction)didTapGlutenFree:(id)sender {
+    [self refreshButtons:self.glutenFreeButton];
+}
+
+- (IBAction)didTapKosher:(id)sender {
+    [self refreshButtons:self.kosherButton];
+}
+
+- (IBAction)didTapVegetarian:(id)sender {
+    [self refreshButtons:self.vegitarianButton];
+}
+
+- (IBAction)didTapVegan:(id)sender {
+    [self refreshButtons:self.veganButton];
+}
+
+- (IBAction)tapCompleteProfile:(id)sender {
+    SceneDelegate *sceneDelegate = (SceneDelegate *)[UIApplication sharedApplication].connectedScenes.allObjects.firstObject.delegate;
+    
+    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
+    FeedViewController *navViewController = [storyboard instantiateViewControllerWithIdentifier:@"TabNav"];
+    sceneDelegate.window.rootViewController = navViewController;
+    
+    PFObject* currentUser = [PFUser currentUser];
+    
+    if(self.veganButton.selected == YES) {
```

Comment:
Whenever we see any repeated if statements checking multiple things, we want to refactor it to avoid repeating code.

 When the User taps the Complete Profile Button, we want to save all the selected options in the database. One idea is having some sort of State Manager that updates the Profile Values when the user taps on a button (e.g. Kosher). So at any point in time, the State Manager will reflect what the user has selected by that point (e.g Kosher, and Vegetarian) and then when the user hits Complete Profile, instead of checking again which buttons were selected, we just have to send the info in the State Manager to the database.

Reply:
Your state manager could be a dictionary that is updated when each button is tapped.

Reply:
You could condense this all into one function by using CollectionView. I believe it is single selection by default, but there is a property you could change to make it multiselect. Quick example I found: https://www.appcoda.com/ios-collection-view-tutorial/#:~:text=UICollectionView%20supports%20both%20single%20and,set%20the%20property%20to%20YES.